compiled_starters/sqlite-starter-go/dbfile/b_tree_page.go

```diff
@@ -0,0 +1,68 @@
+package dbfile
+
+import "io"
+
+/*
+B-Tree page layout
+
+	// Offset Size Description
+	pageHeader   []byte // 8 or 12 bytes
+	cellPointers []uint16
+	// unallocated space
+	cellsContent []byte
+
+B-Tree page header layout
+
+	flags               uint8
+	firstFreeBlock      uint16
+	cells               uint16
+	cellContent         uint16
+	fragmentedFreeBytes uint8
+	rightmostPointer    uint32 // appears only on interior b-tree pages
+
+*/
+
+type BTreePage struct {
+	page
+	headerSize int
+}
+
+const (
+	IndexFlag = 0x2
+	TableFlag = 0x5
+
+	LeafFlag = 0x8
```

Comment:
Let's have the flag comparisons mirror the structure here + only check for equality, not bit-wise operations: 

<img width="684" alt="Screenshot 2022-10-19 at 23 00 16" src="https://user-images.githubusercontent.com/3893573/196812817-cd67e181-d1b1-44cb-8575-ae8c0b896d40.png">

I'd also try to remove the name "flag" entirely, and instead model "pageType" and have it stored on `BTreePage`. That way we read the byte once, check for comparison and set the page type value right within `ReadBTreePage`. `headerSize` can be computed from `pageType` instead of stored in `BTreePage`.

Reply:
Changed Flags and bit-wise operations. But I'm not sure I understand about headerSize. Do you want to make it a functions which is called instead of int field and which value is calculated each time from pageType, which is its order, is stored in the struct? Why storing pageType is better than storing headerSize?

Reply:
If we want to store only primal data and compute the rest when needed, why do we need to store pageType then?